localstack/services/cloudformation/engine/template_deployer.py

```diff
@@ -1358,13 +1359,31 @@ def apply_change(self, change: ChangeConfig, stack: Stack) -> None:
 
         progress_event = executor.deploy_loop(resource_provider_payload)  # noqa
 
+        # TODO: clean up the surrounding loop (do_apply_changes_in_loop) so that the responsibilities are clearer
+        stack_action = get_action_name_for_resource_change(action)
+        match progress_event.status:
+            case OperationStatus.FAILED:
+                stack.set_resource_status(resource_id, f"{stack_action}_FAILED")
+                # TODO: remove exception raising here?
```

Comment:
currently we still need to raise otherwise the enclosing loop in the template deployer won't properly set the stack/resource states